src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/TemporaryAbsencesResource.kt

```diff
@@ -122,4 +126,63 @@ class TemporaryAbsencesResource(
     @Schema(description = "Prison Number", example = "A1234AA", required = true)
     @PathVariable prisonNumber: String
   ): TemporaryAbsence = temporaryAbsenceService.getTemporaryAbsence(agencyId, prisonNumber)
+
+  @PreAuthorize("hasRole('ROLE_VIEW_ARRIVALS')")
+  @Operation(
+    summary = "Confirm return from temporary absence",
+    description = "Confirm return from temporary absence, role required is ROLE_VIEW_ARRIVALS",
+    security = [SecurityRequirement(name = "ROLE_VIEW_ARRIVALS", scopes = ["write"])],
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Confirm return from temporary absence",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = InmateDetail::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to retrieve",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Temporary absence data not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "500",
+        description = "Unexpected error",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+    ]
+  )
+  @PostMapping(
+    "/prisoner/{prisonNumber}/temporary-absence-arrival",
+    consumes = [MediaType.APPLICATION_JSON_VALUE],
+    produces = [MediaType.APPLICATION_JSON_VALUE]
+  )
+  fun temporaryAbsencesArrival(
+    @PathVariable
+    prisonNumber: String,
+
+    @RequestBody
+    temporaryAbsencesArrival: TemporaryAbsencesArrival
+  ) {
+
+    temporaryAbsenceService.temporaryAbsencesArrival(prisonNumber, temporaryAbsencesArrival)
```

Comment:
meh.. could call this `confirmTemporaryAbsencesArrival` ?

Reply:
done